src/Generation/GapicClientV2Generator.php

```diff
@@ -507,29 +507,31 @@ private function construct(): PhpClassMember
         $optionsParam = AST::param(ResolvedType::array(), $options, AST::array([]));
         $clientOptions = AST::var('clientOptions');
 
-        // Assumes there are only two transport types.
-        $isGrpcRest = $this->serviceDetails->transportType === Transport::GRPC_REST;
-
-        $restTransportDocText = 'At the moment, supports only `rest`.';
-        $grpcTransportDocText = 'May be either the string `rest` or `grpc`. Defaults to `grpc` if gRPC support is detected on the system.';
         $transportDocText =
             PhpDoc::text(
                 'The transport used for executing network requests. ',
-                $isGrpcRest ? $grpcTransportDocText : $restTransportDocText,
+                $this->transportDocText(),
                 '*Advanced usage*: Additionally, it is possible to pass in an already instantiated',
                 // TODO(vNext): Don't use a fully-qualified type here.
                 $ctx->type(Type::fromName(TransportInterface::class), true),
                 'object. Note that when this object is provided, any settings in $transportConfig, and any $apiEndpoint',
                 'setting, will be ignored.'
             );
+        
+        
+        $transportType = $this->serviceDetails->transportType;
+        $transportConfigSampleValues = [
+            'grpc' => AST::arrayEllipsis(),
+            'rest' => AST::arrayEllipsis()
+        ];
 
-        $transportConfigSampleValues = [];
-        if ($isGrpcRest) {
-            $transportConfigSampleValues['grpc'] = AST::arrayEllipsis();
+        if (Transport::isGRPCOnly($transportType)) {
+            unset($transportConfigSampleValues['rest']);
+        }
+        if (Transport::isRestOnly($transportType)) {
```

Comment:
mutually exclusive, should be `elseif`

```suggestion
        } elseif (Transport::isRestOnly($transportType)) {
```

Reply:
Changed it.

I just personally do not really like the use of `else` and `elseif`. Is jut a nitpick of mine 😅